src/general/EndosymbiosisData.cs

```diff
@@ -0,0 +1,274 @@
+    ///     TODO: determine if it would be better to have this be a list to allow multiple endosymbiosis operations
+    ///     at once
+    ///   </para>
+    /// </remarks>
+    [JsonProperty]
+    public InProgressEndosymbiosis? StartedEndosymbiosis { get; private set; }
+
+    /// <summary>
+    ///   Completed endosymbiosis operations
+    /// </summary>
+    [JsonProperty]
+    public List<Endosymbiont>? Endosymbionts { get; private set; }
+
+    public OrganelleDefinition GetOrganelleTypeForInProgressSymbiosis(Species symbiontSpecies)
+    {
+        if (StartedEndosymbiosis == null)
+            throw new InvalidOperationException("No in-progress endosymbiosis operations, can't find type");
+
+        if (StartedEndosymbiosis.Species != symbiontSpecies)
+            throw new ArgumentException("Specified species is not an in-progress endosymbiont");
+
+        return StartedEndosymbiosis.TargetOrganelle;
+    }
+
+    /// <summary>
+    ///   Increments progress by one for the given species
+    /// </summary>
+    /// <returns>True on success</returns>
+    public bool ReportEndosymbiosisProgress(Species symbiontSpecies)
+    {
+        if (StartedEndosymbiosis == null)
+            return false;
+
+        if (StartedEndosymbiosis.Species != symbiontSpecies)
+        {
+            GD.PrintErr("Endosymbiont to update progress for doesn't match the in-progress endosymbiosis operation");
+            return false;
+        }
+
+        StartedEndosymbiosis.CurrentlyAcquiredCount += 1;
+
+        // Clamp to max to for example prevent progress indicators from showing more than full
+        if (StartedEndosymbiosis.CurrentlyAcquiredCount > StartedEndosymbiosis.RequiredCount)
+            StartedEndosymbiosis.CurrentlyAcquiredCount = StartedEndosymbiosis.RequiredCount;
+
+        return true;
+    }
+
+    public bool StartEndosymbiosis(int targetSpecies, OrganelleDefinition organelle, int cost)
+    {
+        if (StartedEndosymbiosis != null)
+        {
+            GD.PrintErr("Endosymbiosis is already in progress");
+            return false;
+        }
+
+        if (cost < 1)
+            throw new ArgumentException("Cost should be at least one for endosymbiosis");
+
+        // Need to resolve the species before can start
+        foreach (var candidate in EngulfedSpecies)
+        {
+            if (candidate.Key.ID != targetSpecies || candidate.Value <= 0)
+                continue;
+
+            if (Endosymbionts != null)
+            {
+                foreach (var endosymbiont in Endosymbionts)
+                {
+                    if (endosymbiont.OriginallyFromSpecies.ID == targetSpecies)
+                    {
+                        GD.PrintErr("Trying to form symbiosis with something that is already an endosymbiont");
+                        return false;
+                    }
+                }
+            }
+
+            StartedEndosymbiosis = new InProgressEndosymbiosis(candidate.Key, cost, organelle);
```

Comment:
I'm probably wrong, but this is never discarded, so...

Reply:
What do you mean by discarded? This value is stored in the property and this method next returns true so the value will stay there until cancelled or completed.

Reply:
> What do you mean by discarded?
Disposed. You know, it was created with `new` but isn't destroyed anywhere (I saw that `StartedEndosymbiosis` is being set to null somewhere though). But as I said, I am probably wrong here...

Reply:
`InProgressEndosymbiosis` is not an `IDisposable` so it is not possible to even dispose it. As it is a plain C# class that just uses memory and references other similar objects, the garbage collector takes care of deallocating it when it is no longer used (which is kind of the point with garbage collected programming languages like C# versus C++).